frontend/pages/DashboardPage/DashboardPage.tsx

```diff
@@ -428,7 +460,74 @@ const DashboardPage = ({ router, location }: IDashboardProps): JSX.Element => {
     setShowAddHostsModal(!showAddHostsModal);
   };
 
-  const { MANAGE_HOSTS } = paths;
+  // NOTE: this is called once on the initial rendering. The initial render of
+  // the TableContainer child component will call this handler.
+  const onSoftwareQueryChange = async ({
+    pageIndex: newPageIndex,
+  }: ITableQueryData) => {
+    if (softwarePageIndex !== newPageIndex) {
+      setSoftwarePageIndex(newPageIndex);
+    }
+  };
+
+  const onSoftwareTabChange = (index: number) => {
+    const { SOFTWARE_TITLES } = paths;
+    setSoftwareNavTabIndex(index);
+    setSoftwareActionUrl &&
+      setSoftwareActionUrl(
+        index === 1 ? `${SOFTWARE_TITLES}?vulnerable=true` : SOFTWARE_TITLES
+      );
+  };
+
+  const onSubmitActivityFeedAutomationsModal = useCallback(
```

Comment:
this is the only new code here – `onSoftwarequeryChange`, `onSoftwareTabChange`, and `isCollectingInventory` are just moved around